content/terraform/v1.12.x/docs/language/parameterize/configure.mdx

```diff
@@ -0,0 +1,428 @@
+
+```hcl
+locals {
+  # Naming convention
+  resource_name = "${var.project_name}-${var.environment}"
+
+  # Process the subnet list
+  primary_public_subnet = var.subnet_ids[0]
+  subnet_count          = length(var.subnet_ids)
+
+  # Environmental deployment settings
+  is_production      = var.environment == "prod"
+  monitoring_enabled = var.monitoring || local.is_production
+}
+```
+
+</CodeBlockConfig>
+
+To reference the values from the `locals` block in your configuration, use `local.<NAME>` syntax. The `locals` block defines local values, and you reference those values on an object named `local`, which is singular.
+
+For example, in your configuration you can reference `local.resource_name` to name resources, and use `local.primary_public_subnet` with `local.monitoring_enabled` to configure a web server:
+
+<CodeBlockConfig highlight="4,5,8,14,17" hideClipboard>
+
+```hcl
+resource "aws_instance" "web" {
+  ami           = data.aws_ami.ubuntu.id
+  instance_type = var.instance_type
+  subnet_id     = local.primary_public_subnet
+  monitoring    = local.monitoring_enabled
+
+  tags = {
+    Name        = local.resource_name
+    Environment = var.environment
+  }
+}
+
+resource "aws_security_group" "web" {
+  name = "${local.resource_name}-sg"
+
+  tags = {
+    Name = "${local.resource_name}-security-group"
+  }
+}
+```
+
+</CodeBlockConfig>
+
+You can access local values in the module where you define them, but not in other modules. However, you can pass a local value to a child module as an [argument](/terraform/language/modules/syntax#calling-a-child-module).
+
+Local values help avoid repetition and give a meaningful name to the value of an expression. However, they can make configuration harder to read because they obscure where values originate. Use local values in situations where you reuse a single value in many places to let you change a value in a single place, or when the value is the result of a complex expression.
+
+To learn more about the `locals` block, refer to the [`locals` block reference](/terraform/language/block/locals).
+
+## Define outputs to expose module data
+
+> **Hands-on:** Try the [Output data from Terraform](/terraform/tutorials/configuration-language/outputs) tutorial.
+
+Outputs let you expose information about your infrastructure on the command line, in HCP Terraform, and in other Terraform configurations. The `output` block serves four main purposes in Terraform:
+
+- Child modules can expose resource attributes to parent modules.
+- Root modules can display values in CLI output.
+- Other Terraform configurations using [remote state](/terraform/language/state/remote) can access root module outputs with the `terraform_remote_state` data source.
+- Pass information from a Terraform operation to an automation tool.
+
+Add `output` blocks to export information about your module. For example, you can add two `output` blocks to expose the ID and the IP address of your configuration's web server:
```

Comment:
```suggestion
Add `output` blocks to export information about your module. For example, you can add two `output` blocks to expose the ID and the IP address of your configuration's web server:
```
I've made similar suggestions elsewhere, but it would be more concrete to describe what's happening "in the following example" and to refer to the names of the arguments and keywords being used to create the examples.

Reply:
I think folks understand that `instance_id` and `instance_ip` map to " you can add two `output` blocks to expose the ID and the IP address of your configuration's web server" here.